### PR TITLE
Make pubinternal types in Antiforgery internal

### PR DESCRIPTION
**Pull request by pranavkm**
Adresses https://github.com/aspnet/AspNetCore/issues/8308

### CODE REVIEW

**Comment by pranavkm** [comment]
Pre-emptively sending out a PR. Most of these types fall in to the category of default implementations of interfaces, so it makes it fairly easy to consider making them true `internal`. `IAntiforgeryFeature` was the one type that stuck out. It's set in `FeatureCollection` but only consumed by `Antiforgery`. Should it be made `public` since it's an actual usable interface?

File: src/Antiforgery/src/Microsoft.AspNetCore.Antiforgery.csproj [hunk 1]
```diff
@@ -1,10 +1,9 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <Description>An antiforgery system for ASP.NET Core designed to generate and validate tokens to prevent Cross-Site Request Forgery attacks.</Description>
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <IsAspNetCoreApp>true</IsAspNetCoreApp>
-    <NoWarn>$(NoWarn);CS1591</NoWarn>
```
Comment (Eilon) [inline src/Antiforgery/src/Microsoft.AspNetCore.Antiforgery.csproj]: NICE!

**Review by Eilon — APPROVED** [review]

**Comment by pranavkm** [comment]
Spoke to @rynowak offline. `IAntiforgeryFeature` seems like an implementation detail and doesn't need to be public. We can wait for user feedback to see if it needs to be made public.

**Comment by logic01** [comment]
```
using Microsoft.AspNetCore.Antiforgery;
            var serializer = afType
                .GetField("_tokenSerializer", BindingFlags.NonPublic | BindingFlags.Instance)
                .GetValue(_antiForgery) as IAntiforgeryTokenSerializer;
```

Looks like my team is one of the few who used this interface in dotnet core 2.x. Is there any way to get similar functionality?

**Pull request merged**